### PR TITLE
feat(pagination): hide 1 dot pagination

### LINKED ISSUE #2572: Request - hide pagination dots if only 1 "page"
This is a (multiple allowed):
* [] bug
* [x] enhancement
* [x] feature-discussion (RFC)

* Swiper Version: 4.2.2.
* Platform/Target and Browser Versions: ALL.
* Live Link or JSFiddle/Codepen or website with issue: N/A

### What you did
* Created a multi-row swiper which has it's contents dynamically updated (slides added/removed).
* On initial load, only 1 "page" may have slides, so the pagination button is a distraction.
* Swipers may or may not have slides dynamically added as the user interacts with the application

### Expected Behavior
* If a swiper contains only 1 "page", dynamically hide the pagination dots.
* If more slides are dynamically added (e.g lazy loaded) and this results in total "pages" > 1, dynamical add the pagination buttons.
* If total "pages" > 1 and slides are destroyed, evaluate and dynamically hide the pagination dots.

### Actual Behaviour
If pagination is enabled, page dots are shown, even if only 1 "page" is visible (aka no swiping) - this decreases the usability and is likely to confuse end users - after all, pagination is only useful if there are multiple pages 😃

### PR DESCRIPTION
**Pull request by vltansky**
closes #2572

### CODE REVIEW

File: src/components/pagination/pagination.less [hunk 1]
```diff
@@ -77,6 +77,10 @@
   .swiper-pagination-clickable & {
     cursor: pointer;
   }
+
+  &:only-child {
+    display: none;
```
Comment (nolimits4web) [inline src/components/pagination/pagination.less:82]: `display: none` here

File: src/components/pagination/pagination.scss [hunk 2]
```diff
@@ -67,6 +67,10 @@
   .swiper-pagination-clickable & {
     cursor: pointer;
   }
+
+  &:only-child {
+    visibility: hidden;
```
Comment (nolimits4web) [inline src/components/pagination/pagination.scss:72]: `visibility: hidden;` here? 😀

**Review by nolimits4web — CHANGES_REQUESTED** [review]
I that is so simple :D let's make it for v6 then :)

File: src/components/pagination/pagination.scss (hunk 2 above)
Comment (vltansky) [reply src/components/pagination/pagination.scss]: you're right

**Comment by vltansky** [comment]
closed in favor of #4786 (that target `master` instead of `v7`)

**Pull request closed without merging**